code/Model/EditableFormField.php

```diff
@@ -442,6 +443,11 @@ public function onBeforeWrite()
                 ->filter('ParentID', $parentID)
                 ->max('Sort') + 1;
         }
+
+        // Include _new_ Email Fields by default in GridField summary (not possible via $defaults since the initial record is always EditableFormField)
+        if ($this->isChanged('ClassName') && $this->ClassName ===  EditableEmailField::class) {
+            $this->ShowInSummary = true;
+        }
```

Comment:
Should this go into `EditableEmailField`'s `onBeforeWrite` then?

Reply:
I can't think of a way to detect the 'init' of a new `EditableEmailField` in that onBeforeWrite. Since the object is already created in DB, and changed to Email when the object already exists... But if you have suggestions?

Reply:
Moving this exact code there doesn't work?

Reply:
Correct, that doesn't work. Just double checked it...

The point is that the onBeforeWrite is only triggered on the Class EditableTextField the first time. Once it is saved, the second time it does trigger the onBeforeWrite, but than it is not detectable if it was a new field...

Maybe using onAfterWrite solves this problem? But setting a field after writing, and not getting into a infinite loop is always rather cumbersome.

Reply:
Do you need to know if it's a new field? I thought you needed to know if it was an email field and set the ShowInSummary to true for those. Doesn't matter if it's a new field or an existing one, checking the ClassName field whether it changed or already is `EditableEmailField::class` should be enough? Or am I missing something?

Reply:
Yes, we would like to let email-addresses to show up in the summary by default. But give the editor the option to remove it from the summary, if that what he/she wants. 

Ideally one would use ‘private static $defaults’ for this. However since it is never a EmailField that is created, this seems the only reasonable alternative.